ts/input/tex/mathtools/MathtoolsMethods.ts

```diff
@@ -0,0 +1,747 @@
+  },
+
+  /**
+   * Implements \DeclarePairedDelimitersXPP.
+   *
+   * @param {TexParser} parser   The calling parser.
+   * @param {string} name        The macro name.
+   */
+  DeclarePairedDelimitersXPP(parser: TexParser, name: string) {
+    const cs = NewcommandUtil.GetCsNameArgument(parser, name);
+    const n = NewcommandUtil.GetArgCount(parser, name);
+    const pre = parser.GetArgument(name);
+    const open = parser.GetArgument(name);
+    const close = parser.GetArgument(name);
+    const post = parser.GetArgument(name);
+    const body = parser.GetArgument(name);
+    MathtoolsUtil.addPairedDelims(parser.configuration, cs, [open, close, body, n, pre, post]);
+  },
+
+  /**
+   * Implements \centeredcolon, \ordinarycolon, \MTThinColon.
+   *
+   * @param {TexParser} parser   The calling parser.
+   * @param {string} name        The macro name.
+   * @param {boolean} center     True if colon should be centered
+   * @param {boolean} force      True menas always center (don't use centercolon option).
+   * @param {boolean} thin       True if this is a thin color (for \coloneqq, etc).
+   */
+  CenterColon(parser: TexParser, _name: string, center: boolean, force: boolean = false, thin: boolean = false) {
+    const options = parser.options.mathtools;
+    let mml = parser.create('token', 'mo', {}, ':');
+    if (center && (options['centercolon'] || force)) {
+      const dy = options['centercolon-offset'];
+      mml = parser.create('node', 'mpadded', [mml], {
+        voffset: dy, height: `+${dy}`, depth: `-${dy}`,
+          ...(thin ? {width: options['thincolon-dw'], lspace: options['thincolon-dx']} : {})
+      });
+    }
+    parser.Push(mml);
+  },
+
+  /**
+   * Implements \coloneqq and related macros.
+   *
+   * @param {TexParser} parser   The calling parser.
+   * @param {string} name        The macro name.
+   * @param {string} tex         The tex string to use (if not using unicode versions or if there isn't one).
+   * @param {string} unicode     The unicode character (if there is one).
+   */
+  Relation(parser: TexParser, _name: string, tex: string, unicode?: string) {
+    const options = parser.options.mathtools;
+    if (options['use-unicode'] && unicode) {
+      parser.Push(parser.create('token', 'mo', {texClass: TEXCLASS.REL}, unicode));
+    } else {
+      tex = '\\mathrel{' + tex.replace(/:/g, '\\MTThinColon').replace(/-/g, '\\mathrel{-}') + '}';
+      parser.string = ParseUtil.addArgs(parser, tex, parser.string.substr(parser.i));
+      parser.i = 0;
+    }
+  },
+
+  /**
+   * Implements \ndownarrow and \nuparrow via a terrible hack (visual only, no chance of this working with SRE).
+   *
+   * @param {TexParser} parser   The calling parser.
+   * @param {string} name        The macro name.
+   */
+  NArrow(parser: TexParser, _name: string, c: string, dy: string) {
+    parser.Push(
+      parser.create('node', 'TeXAtom', [
+        parser.create('token', 'mtext', {}, c),
```

Comment:
What is `c`?

Reply:
It's the unicode character for either an up arrow or a down arrow (as provided by the `MathtoolsMappings.ts` file).